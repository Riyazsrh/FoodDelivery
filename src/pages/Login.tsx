import { Button, Grid, ImageList, TextField, Typography, Container, MenuItem, Select } from '@mui/material'
import { Box } from '@mui/system'
import { useStyles } from './Loginstyles'
import Ellips from '../images/Ellipse 14.png'
import { useNavigate } from 'react-router-dom'
import LocalPhoneIcon from '@mui/icons-material/LocalPhone';
import { useEffect, useState } from 'react'
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';

const Login = () => {
  const [username, setUserName] = useState<any>("");
  const navigate = useNavigate();
  const classes = useStyles();

  useEffect(() => {
    if (sessionStorage.getItem('logindata')) {
      navigate('/menu')
    }
  })

  const PostData = async () => {

    const Data: any = {
      username: username,
    }

    await fetch("https://extended-retail-app.herokuapp.com/api/customer/getOtp",
      {
        method: "POST",
        headers: {
          Accept: "application/json",
          "Content-Type": "application/json",
        },
        body: JSON.stringify({ username: username }),
      }
    )
      .then((resp) => resp.json())
      .then((respo) => {
        console.log("result", respo);
        if (respo.status === 200) {
          localStorage.setItem("data", JSON.stringify(respo));
          localStorage.setItem("mob", username);
          toast('OTP sent to your mobile number')
          navigate(`/verification/${username}`)
        } else {
          toast('Please Enter valid Number ‚ùå')
        }
      });
  }



  return (
    <Container maxWidth="lg">
      <Grid container minHeight="100vh" direction="row" columns={{ xs: 4, sm: 10, md: 12 }} alignItems="center" justifyContent="center">
        <Grid item xs={5} sm={5} md={5} lg={5} >
          <Box sx={{ display: "flex", flexDirection: "column", alignItems: "center" }}>
            <Box >
              <img src='./images/logo 1.png' alt='logo' data-testid="logoImg" />
            </Box>
            <Box >
              <Typography variant="h3" sx={{ fontSize: "35px", fontFamily: "Bai Jamjuree" }} gutterBottom component="div">
                Welcome Back!
              </Typography>
            </Box>
            <Box >
              <Typography variant="h6" sx={{ color: "#A2A3A5" }} gutterBottom component="div">
                Login Account
              </Typography>
            </Box>
            <Box sx={{ display: "flex" }}>
              <Select
                labelId="demo-simple-select-label"
                id="demo-simple-select"
                sx={{ height: "50px" }}
                label="Select"
              >
                <MenuItem value={10}><img src='images\1200px-Flag_of_India 1.png' style={{ marginTop: "5px" }} />+91</MenuItem>
              </Select>
              <TextField data-testid="loginInput" id="standard-basic" value={username} onChange={(e) => { setUserName(e.target.value) }} sx={{ width: "270px", height: "50px" }} placeholder="mobile no" label="mobile no" variant="standard" InputProps={{
                endAdornment: (
                  <LocalPhoneIcon />
                ),
              }} />
            </Box>
            <Button data-testid="loginButton" disabled={username.length < 10} className={username.length < 10 ? classes.otp1 : classes.otp} style={{ margin: "30px 0" }} onClick={PostData} >GET OTP</Button>
          </Box>
        </Grid>

        <Grid item xs={7} sm={7} md={7} lg={7}>
          <Box sx={{ backgroundImage: `url(${Ellips})` }}>
            <img src='../images/img.png' alt='bike' style={{ width: "100%", backgroundRepeat: "no-repeat" }} />
          </Box>

        </Grid>
      </Grid>
      <ToastContainer position="top-center" />
    </Container >












  )
}

export default Login
