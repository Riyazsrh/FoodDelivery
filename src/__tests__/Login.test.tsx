import { findByText, fireEvent, render, screen } from "@testing-library/react"
import Login from "../pages/Login"
import renderer from "react-test-renderer";


const mockUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => mockUsedNavigate,
}));


beforeEach(() => {
    global.fetch = jest.fn(() => {
        return Promise.resolve({
            json: () =>
                Promise.resolve({
                    status: 200, data: {
                        nbPages: 2, hits: [{
                            id: '1234',
                            title: "getOtp",
                            url: "https://extended-retail-app.herokuapp.com/api/customer/getOtp",
                            created_at: "CreatedAt",
                            author: "Author",
                        }]
                    }
                }),
        });
    }) as jest.Mock;
})

test("Login Logo Image test", () => {
    const doc = render(<Login />);
    const logo = doc.getByTestId("logoImg");
    expect(logo).toBeInTheDocument();
})

test("Login Text test", () => {
    const doc = render(<Login />);
    const textCheck = doc.getByText(/Welcome Back!/i);
    const textCheck2 = doc.getByText(/Login Account/i);

    expect(textCheck).toBeInTheDocument();
    expect(textCheck2).toBeInTheDocument();
})


test("Login Input Field test", () => {
    const doc = render(<Login />);
    const log = doc.getByTestId("loginInput");
    expect(log).toBeInTheDocument();
})


test("Login Input Field values test", () => {
    const doc = render(<Login />);
    const inputElement = doc.getByTestId("loginInput");

    expect(inputElement).toBeInTheDocument();
})



it("should validate mobile numbers", () => {

    const { getByPlaceholderText } = render(<Login />);
    const inputCheck: any = getByPlaceholderText("mobile no");

    fireEvent.change(inputCheck, {
        target: { value: "" }
    });
    expect(inputCheck.value).toBe('');
});


test("get otp button", () => {
    render(<Login />)
    const button = screen.getByTestId("loginButton")

    fireEvent.click(button);
    expect(button).toBeInTheDocument();

})



test("api testing", () => {
    const UserResponse = { token: null }

    jest.spyOn(window, 'fetch').mockImplementationOnce((): any => {
        return Promise.resolve({
            json: () => Promise.resolve(UserResponse),
        })
    })
    expect(window.localStorage.getItem('token')).toBe(UserResponse.token)
})

it("Login snnapshot", () => {
    const tree = renderer.create(<Login />)
    expect(tree).toMatchSnapshot();
})
