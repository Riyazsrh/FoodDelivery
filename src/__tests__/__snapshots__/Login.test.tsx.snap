// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snnapshot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
>
  <div
    className="MuiGrid-root MuiGrid-container css-p08t7n-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 MuiGrid-grid-sm-5 MuiGrid-grid-md-5 MuiGrid-grid-lg-5 css-1qr3d4p-MuiGrid-root"
    >
      <div
        className="MuiBox-root css-dvxtzn"
      >
        <div
          className="MuiBox-root css-0"
        >
          <img
            alt="logo"
            data-testid="logoImg"
            src="./images/logo 1.png"
          />
        </div>
        <div
          className="MuiBox-root css-0"
        >
          <div
            className="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-13gc1yq-MuiTypography-root"
          >
            Welcome Back!
          </div>
        </div>
        <div
          className="MuiBox-root css-0"
        >
          <div
            className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-3zk4c-MuiTypography-root"
          >
            Login Account
          </div>
        </div>
        <div
          className="MuiBox-root css-k008qs"
        >
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary  css-1nreaws-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            onClick={[Function]}
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-label demo-simple-select"
              className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="demo-simple-select"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="notranslate"
              >
                â€‹
              </span>
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              tabIndex={-1}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Select
                </span>
              </legend>
            </fieldset>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root css-1wtapkw-MuiFormControl-root-MuiTextField-root"
            data-testid="loginInput"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="standard-basic"
              id="standard-basic-label"
            >
              mobile no
            </label>
            <div
              className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1ptx2yq-MuiInputBase-root-MuiInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                disabled={false}
                id="standard-basic"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="mobile no"
                required={false}
                type="text"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LocalPhoneIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
                />
              </svg>
            </div>
          </div>
        </div>
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-otp-33 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="loginButton"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "margin": "30px 0",
            }
          }
          tabIndex={0}
          type="button"
        >
          GET OTP
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 MuiGrid-grid-sm-7 MuiGrid-grid-md-7 MuiGrid-grid-lg-7 css-1u5iry1-MuiGrid-root"
    >
      <div
        className="MuiBox-root css-1iautga"
      >
        <img
          alt="bike"
          src="../images/img.png"
          style={
            Object {
              "backgroundRepeat": "no-repeat",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
